File: src/main/kotlin/sunsetsatellite/vm/sunlite/SLBoundMethod.kt
package sunsetsatellite.vm.sunlite

class SLBoundMethod(val method: SLClosure, val receiver: AnySLValue) {

	override fun toString(): String {
		return method.toString()
	}

	fun copy(): SLBoundMethod {
		return SLBoundMethod(method.copy(), receiver.copy())
	}
}

File: src/main/csharp/SunliteSharp/Runtime/SLBoundMethod.cs
namespace SunliteSharp.Runtime;

public record SLBoundMethod(SLClosure Method, AnySLValue Receiver)
{
    public override string ToString()
    {
        return Method.ToString();
    }

    public SLBoundMethod Copy()
    {
        return new SLBoundMethod(Method.Copy(), Receiver.Copy());
    }
}